flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/rules/physical/stream/StreamPhysicalPythonGroupWindowAggregateRule.java

```diff
@@ -98,7 +98,12 @@ public RelNode convert(RelNode rel) {
         FlinkLogicalWindowAggregate agg = (FlinkLogicalWindowAggregate) rel;
         LogicalWindow window = agg.getWindow();
 
-        if (window instanceof SessionGroupWindow) {
+        List<AggregateCall> aggCalls = agg.getAggCallList();
+        boolean isPandasPythonUDAF =
+                aggCalls.stream()
+                        .anyMatch(x -> PythonUtil.isPythonAggregate(x, PythonFunctionKind.PANDAS));
+
+        if (isPandasPythonUDAF && window instanceof SessionGroupWindow) {
             throw new TableException("Session Group Window is currently not supported.");
```

Comment:
Nit: probably `"Session Group Window is currently not supported for Pandas UDAF"`.

Also why session window does not work when it is Pandas UDAF?

Reply:
This is only temporarily not supported, and there are design solutions for this part in [FLIP-137.](https://cwiki.apache.org/confluence/display/FLINK/FLIP-137%3A+Support+Pandas+UDAF+in+PyFlink)